File: common/src/main/java/guepardoapps/lucahome/common/classes/ILucaClass.java
package guepardoapps.lucahome.common.classes;

import android.support.annotation.NonNull;

import java.io.Serializable;
import java.util.UUID;

public interface ILucaClass extends Serializable {
    enum LucaServerDbAction {Null, Add, Update, Delete}

    UUID GetUuid();

    UUID GetRoomUuid() throws NoSuchMethodException;

    String GetCommandAdd() throws NoSuchMethodException;

    String GetCommandUpdate() throws NoSuchMethodException;

    String GetCommandDelete() throws NoSuchMethodException;

    void SetIsOnServer(boolean isOnServer) throws NoSuchMethodException;

    boolean GetIsOnServer() throws NoSuchMethodException;

    void SetServerDbAction(@NonNull LucaServerDbAction lucaServerDbAction) throws NoSuchMethodException;

    LucaServerDbAction GetServerDbAction() throws NoSuchMethodException;
}

File: common/src/main/java/guepardoapps/lucahome/common/models/common/ILucaClass.kt
package guepardoapps.lucahome.common.models.common

import guepardoapps.lucahome.common.enums.ServerDatabaseAction

import java.io.Serializable
import java.util.UUID

interface ILucaClass : Serializable {
    val uuid: UUID

    val commandAdd: String
    val commandUpdate: String
    val commandDelete: String

    var isOnServer: Boolean
    var serverDatabaseAction: ServerDatabaseAction
}
